torch/nn/functional.py

```diff
@@ -3077,3 +3077,197 @@ def _pad_circular(input, padding):
         input = torch.cat([input[:, :, :, :, -(padding[-5] + padding[-6]):-padding[-5]], input], dim=4)
 
     return input
+
+
+@weak_script
+def multi_head_attention_forward(
+        query, key, value, embed_dim_to_check, num_heads,
+        in_proj_weight, in_proj_bias, bias_k, bias_v, add_zero_attn,
+        dropout_p, out_proj, training=True,
+        key_padding_mask=None, need_weights=True, attn_mask=None):
+    # type: (Tensor, Tensor, Tensor, int, int, Tensor, Tensor, Tensor, Tensor, bool, float, Tensor, bool, Optional[Tensor], bool, Optional[Tensor]) -> Tuple[Tensor, Tensor]
```

Comment:
This shows up [in the CI](https://travis-ci.org/pytorch/pytorch/jobs/531948682#L234) as too long.